src/DynamoRevit/DynamoRevit.cs

```diff
@@ -214,8 +235,13 @@ private static DynamoViewModel InitializeCoreViewModel(RevitDynamoModel revitDyn
 
             revitDynamoModel.ShutdownStarted += (drm) =>
             {
-                var uiApplication = DocumentManager.Instance.CurrentUIApplication;
-                uiApplication.Idling += DeleteKeeperElementOnce;
+                lock (actions)
+                {
+                    DynamoRevit.actions.Add(DeleteKeeperElement);
+                }
+
+                // var uiApplication = DocumentManager.Instance.CurrentUIApplication;
+                // uiApplication.Idling += DeleteKeeperElementOnce;
```

Comment:
The `DeleteKeeperElementOnce` method should be removed, and this method updated to just the following:

```csharp
revitDynamoModel.ShutdownStarted += (drm) =>
{
    lock (idleActions)
    {
        idleActions.Add(DeleteKeeperElement);
    }
};
```